File: src/java/lecture/src/_03_classes/d_static/Circle.java
package _03_classes.d_static;

/** Lecture sample code.<br><br>
 *
 * Author: Marc Hensel (<a href="http://www.haw-hamburg.de/marc-hensel">link</a>)<br>
 * Project: coding_learners_java (<a href="https://github.com/MarcOnTheMoon/coding_learners_java/">link</a>)<br>
 * Copyright: 2024, Marc Hensel<br>
 * Version: 2024.01.30<br>
 * License: CC BY-NC-SA 4.0 (see <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en">link</a>)
 */
public class Circle {
	double x, y, radius;
	static final double PI = 3.141592653589793;
	static int count;
	
	Circle() {
		radius = 1.0;
		count++;
	}

	static double getPi() {
		return 3.141592653589793;
	}
	
	double getArea() {
		return PI * radius * radius;
	}
}

File: src/c/lecture/03_FlowControl/01-04_if_Circle/circle.c
/*****************************************************************************************************
 * Lecture sample code.
 * Calculate circumference and area of a circle, only if the radius is non-negative.
 *****************************************************************************************************
 * Author: Marc Hensel, http://www.haw-hamburg.de/marc-hensel
 * Project: https://github.com/MarcOnTheMoon/coding_learners_c/
 * Copyright: 2023, Marc Hensel
 * Version: 2023.08.23
 * License: CC BY-NC-SA 4.0, see https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en
 *****************************************************************************************************/

#define _CRT_SECURE_NO_DEPRECATE	// Else MSVC++ prevents using scanf() (concern: buffer overflow)
#include <stdio.h>

int main(void)
{
	float radius;
	const float PI = 3.141592f;

	/* Get user input: radius */
	printf("Please enter a radius: ");
	scanf("%f", &radius);
	getchar();

	/* Print area and circumference to the console */
	if (radius >= 0.0)
	{
		printf("\nRadius       : %.2f units\n", radius);
		printf("Circumference: %.2f units\n", 2.0 * PI * radius);
		printf("Area         : %.2f units^2\n", PI * radius * radius);
	}
	else
		printf("\nThere is no circle with negative radius.");

	getchar();
	return 0;
}
